### PR TITLE
Linting and style checks

### PR DESCRIPTION
**Pull request by QuantumEntangledAndy**
Mostly to help keep up with rustfmt I have added some git action checks.

I have also applied clippy linting and a check for that too.

### CODE REVIEW

**Comment by QuantumEntangledAndy** [comment]
Clippy-check action won't work fully until it merged (My token dosent have permission to add annotations to your repo) but it should annotate future PRs like [this](https://github.com/actions-rs/example/pull/2/files)

File: .github/workflows/style_checks.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  check_fmt:
+    name: Rust-fmt
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions-rs/toolchain@v1
+      with:
+          toolchain: nightly
+          components: rustfmt
+          override: true
```
Comment (thirtythreeforty) [inline .github/workflows/style_checks.yaml:30]: Why does this use the nightly toolchain? I try to stick to stable as much as I can.

Comment (thirtythreeforty) [reply .github/workflows/style_checks.yaml:30]: (Sometimes it's needed for fancy features, I just don't know in this case)

Comment (QuantumEntangledAndy) [reply .github/workflows/style_checks.yaml:30]: If I run rustfmt on stable it complains that stable rustfmt is deprecated and instructions me to use nightly. I think perhaps the newer versions of rustfmt require features in nightly. This is not our code on nightly but rather the latest rustfmt is supposed to be on nightly.

Comment (thirtythreeforty) [reply .github/workflows/style_checks.yaml:30]: Okay, fair enough. I wish they'd be making _opposite_ progress (deprecating nightly and working to get it working on stable) but I don't mind quite so much for dev tooling.

**Review by thirtythreeforty — APPROVED** [review]

**Comment by thirtythreeforty** [comment]
@QuantumEntangledAndy the Clippy check is failing even after merge:

```
    --- stderr
    `"pkg-config" "--libs" "--cflags" "gstreamer-1.0" "gstreamer-1.0 >= 1.14"` did not exit successfully: exit code: 1
    --- stderr
    Package gstreamer-1.0 was not found in the pkg-config search path.
    Perhaps you should add the directory containing `gstreamer-1.0.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'gstreamer-1.0' found
    Package gstreamer-1.0 was not found in the pkg-config search path.
    Perhaps you should add the directory containing `gstreamer-1.0.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'gstreamer-1.0' found
  
  warning: build failed, waiting for other jobs to finish...
```

Would you see if you can figure out why?

**Comment by QuantumEntangledAndy** [comment]
Sure I will look into this first thing tomorrow (it's midnight here). Seems I've just forgotten to get gstreamer installled before the clippy check.

**Comment by QuantumEntangledAndy** [comment]
Just adding this before should work

```yml
- name: apt install gstreamer
        run: |
          sudo apt update
          sudo apt install -y libgstrtspserver-1.0-dev libgstreamer1.0-dev libgstreamer-plugins-bad1.0-dev libgstreamer-plugins-good1.0-dev
```

**Pull request merged**